agent/engine/dockerstate/docker_task_engine_state.go

```diff
@@ -418,3 +445,44 @@ func (state *DockerTaskEngineState) RemoveImageState(imageState *image.ImageStat
 	}
 	delete(state.imageStates, imageState.Image.ImageID)
 }
+
+// SetLables sets labels for a container
+func (state *DockerTaskEngineState) SetLabels(id string, labels map[string]string) {
+	state.lock.Lock()
+	defer state.lock.Unlock()
+
+	state.containerIDToLabels[id] = labels
+}
+
+// GetLabels gets labels by container id
+func (state *DockerTaskEngineState) GetLabels(id string) (map[string]string, bool) {
+	state.lock.RLock()
+	defer state.lock.RUnlock()
+
+	labels, ok := state.containerIDToLabels[id]
+	return labels, ok
```

Comment:
You can just `return state.containerIDToLabels[id]` here.

Reply:
I don't think so. You'll get a compilation error